src/fides/api/service/connectors/query_config.py

```diff
@@ -352,86 +354,123 @@ def format_fields_for_query(
         """
         return [fk.levels[-1] for fk in field_paths]
 
-    def format_clause_for_query(
-        self, string_path: str, operator: str, operand: str
-    ) -> str:
-        """Returns clauses in a format they can be added into SQL queries."""
-        return f"{string_path} {operator} :{operand}"
+    @abstractmethod
+    def format_query_data_name(self, query_data_name: str) -> str:
+        """Returns query_data_name formatted according to specific SQL dialect"""
 
+    @abstractmethod
     def get_formatted_query_string(
         self,
         field_list: str,
         clauses: List[str],
     ) -> str:
-        """Returns an SQL query string."""
-        return f"SELECT {field_list} FROM {self.node.collection.name} WHERE {' OR '.join(clauses)}"
+        """Returns a SQL query string."""
 
-    def get_formatted_update_stmt(
-        self,
-        update_clauses: List[str],
-        pk_clauses: List[str],
+    @abstractmethod
+    def format_clause_for_query(
+        self, string_path: str, operator: str, operand: str
     ) -> str:
-        """Returns a formatted SQL UPDATE statement to fit the Snowflake syntax."""
-        return f"UPDATE {self.node.address.collection} SET {','.join(update_clauses)} WHERE {' AND '.join(pk_clauses)}"
+        """Returs clause formatted in the specific SQL dialect for the query"""
 
-    def generate_query(
+    def generate_query_without_tuples(  # pylint: disable=R0914
```

Comment:
yes, I needed to reuse it but didn't want to mess with the existing inheritance of the classes, and I didn't want to inherit from `QueryStringWithoutTuplesOverrideQueryConfig` since that inherits directly from `SQLQueryConfig` which has things that don't make sense for Scylla